deploy/runtools/run_farm.py

```diff
@@ -39,6 +40,8 @@ class Inst(metaclass=abc.ABCMeta):
         metasimulation_enabled: true if this instance will be running metasimulations
     """
 
+    run_farm: RunFarm
```

Comment:
I don't hink you should provide a naked reference to the RunFarm in `Inst`. That seem overly expansive. Since the termination logic can't be put in `Inst` itself, It would be better to make this reference private and add a `terminate_self` method that uses the private runfarm member to call terminate on itself.

Reply:
Ok I think I addressed this like you recommended.